### ISSUE TITLE
Provide a way to determine which polyfills are required

### ISSUE THREAD

**Issue by aweary** [babel/babel-preset-env#294]
This started from a conversation stemming from this tweet: https://twitter.com/aweary/status/853771117894545408

@hzoo, @JakeChampion and I met today and had a quick discussion about how we could make `babel-preset-env` integrate with https://polyfill.io/. The rough plan we agreed upon was:

1. Move the [use-built-ins plugin](https://github.com/babel/babel-preset-env/blob/2.0/src/use-built-ins-plugin.js) into a standalone package
2. Expose an option to write the list of features requiring a polyfill to disk, instead of including the default `core-js` imports.

This would allow users to read the feature list at build time and include the appropriate polyfill.io link, which uses a query string to communicate which polyfills should be included.

This approach makes polyfill.io integration very easy, but it will also open up new ways for community build tools to fulfill polyfill requirements at build time.

**Comment by hzoo** [comment]
3. (not part of this repo) would be a webpack plugin (apparently like the HtmlWebpackPlugin) that injects the new polyfill.io url as a script tag (someone else should do this)

**Comment by aweary** [comment]
`html-webpack-plugin` does provide [hooks for third party addons](https://github.com/jantimon/html-webpack-plugin#third-party-addons), so it should be easy to write an addon that handles inserting the `script` tag.

**Comment by hzoo** [comment]
Also ref earlier issue I made related to this https://github.com/babel/babel-preset-env/issues/284

**Comment by JakeChampion** [comment]
Started work on extracting out the babel plugin to detect which ES runtime features are being used -- https://github.com/JakeChampion/babel-es-runtime-features-extractor

**Comment by hzoo** [comment]
Cool, although since I'm actively working on the plugin right now I'd rather it just be moved into a package in this repo itself. (Kind of like a monorepo, doesn't have to use lerna).

**Comment by rwjblue** [comment]
I'm also interested in this effort. It would make intelligent polyfill inclusion much much easier in ember-cli land. My immediate need is to have a general purpose utility function that would return the list of required polyfills for the given targets. I think the same could be done for syntax/transforms if we desired.

For a proof of concept, I hacked this together as a proof of concept for ember-cli-babel [here](https://github.com/rwjblue/ember-cli-babel/blob/f3d219c59e031218f4a47a8468a1b5b497fed5fd/index.js#L85-L115). It does function properly, but ideally I wouldn't need to duplicate so much code from babel-preset-env itself.
